collect_app/src/test/java/org/odk/collect/android/projects/ProjectDetailsCreatorTest.kt

```diff
@@ -15,43 +15,87 @@ class ProjectDetailsCreatorTest {
     val projectDetailsCreator = ProjectDetailsCreator(ApplicationProvider.getApplicationContext())
 
     @Test
-    fun `Project name should be generated from url`() {
-        assertThat(projectDetailsCreator.getProject("https://my-project.com").name, `is`("my-project.com"))
-        assertThat(projectDetailsCreator.getProject("https://your-project.com/one").name, `is`("your-project.com"))
-        assertThat(projectDetailsCreator.getProject("http://www.my-project.com").name, `is`("www.my-project.com"))
-        assertThat(projectDetailsCreator.getProject("").name, `is`("Project")) // default project name for invalid urls
-        assertThat(projectDetailsCreator.getProject("something").name, `is`("Project")) // default project name for invalid urls
+    fun `If project name is included in project details should be used`() {
+        assertThat(projectDetailsCreator.createProjectFromDetails(name = "Project X").name, `is`("Project X"))
     }
 
     @Test
-    fun `Project icon should be generated from url`() {
-        assertThat(projectDetailsCreator.getProject("https://my-project.com").icon, `is`("M"))
-        assertThat(projectDetailsCreator.getProject("https://your-project.com/one").icon, `is`("Y"))
-        assertThat(projectDetailsCreator.getProject("http://www.my-project.com").icon, `is`("W"))
-        assertThat(projectDetailsCreator.getProject("").icon, `is`("P")) // default project icon for invalid urls
-        assertThat(projectDetailsCreator.getProject("something").icon, `is`("P")) // default project icon for invalid urls
+    fun `If project name is not included in project details or empty should be generated based on url`() {
+        // Case1: not included
```

Comment:
I think we should pull all these "cases" out in to separate tests

Reply:
You mean never have two cases in one test or what?

Reply:
I just think if we feel like it's worth the comments, then we can just pull them out into separate tests and use the test names instead.

Reply:
Done.